src/AppBundle/Entity/User.php

```diff
@@ -566,4 +573,159 @@ public function setMunicipality($municipality)
     {
         $this->municipality = $municipality;
     }
-}
+
+    /**
+     * @return string
+     */
+    public function getFullName()
+    {
+        return $this->firstName .' '. $this->lastName;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getGenders()
+    {
+        return [
+            self::GENDER_MALE   => 'user.form.gender.m',
+            self::GENDER_FEMALE => 'user.form.gender.f',
+            self::GENDER_X      => 'user.form.gender.x'
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public function getCategoryNames()
+    {
+        $categories = [];
+
+        foreach($this->getCategories() as $category) {
+            $categories[$category->getId()] = $category->getName();
+        }
+
+        foreach($this->getMusicCategories() as $category) {
+            $categories[$category->getId()] = $category->getName();
+        }
+
+        asort($categories);
+
+        return $categories;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCountryName()
+    {
+        return Countries::getName($this->from);
+    }
+
+    /**
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->wantToLearn? 'New': 'Established';
+    }
+
+    /**
+     * @return string
+     */
+    public function getMusicFriendType()
+    {
+        return $this->musicFriend? 'filter.form.music_buddy': 'filter.form.fika_buddy';
+    }
+
+    /**
+     * @return string
+     */
+    public function getFirstConnectionRequestComment()
+    {
+        if ($this->connectionRequests->count()) {
+            return $this->connectionRequests->first()->getComment();
+        }
+
+        return '';
+    }
+
+    /**
+     * @return array
+     */
+    public function getCategoryIds()
+    {
+        $ids = [];
+
+        foreach ($this->categories as $category) {
+            $ids[] = $category->getId();
+        }
+
+        return $ids;
+    }
+
+    /**
+     * @return array
+     */
+    public function getMusicCategoryIds()
+    {
+        $ids = [];
+
+        foreach ($this->musicCategories as $category) {
+            $ids[] = $category->getId();
+        }
+
+        return $ids;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCategoryNameString()
```

Comment:
Put this method in a twig extension (use existing). Inject the translator and you can translate _and_.